sys/include/hashes/sha512.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2023 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_hashes_sha512 SHA-512
+ * @ingroup     sys_hashes_unkeyed
+ * @brief       Implementation of the SHA-512 hashing function
+ * @{
+ *
+ * @file
+ * @brief       Header definitions for the SHA512 hash function
+ *
+ * @author      Mikolai Gütschow <mikolai.guetschow@tu-dresden.de>
+ */
+
+#ifndef HASHES_SHA512_H
+#define HASHES_SHA512_H
+
+#include <inttypes.h>
+#include <stddef.h>
+
+#include "hashes/sha512_common.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Length of SHA512 digests in bytes
+ */
+#define SHA512_DIGEST_LENGTH 64
+
+/**
+ * @brief 1024 Bit (128 Byte) internally used block size for sha512
+ */
+#define SHA512_INTERNAL_BLOCK_SIZE (128)
+
+/**
+ * @brief Context for cipher operations based on sha512
+ */
+typedef sha512_common_context_t sha512_context_t;
+
+
+/**
+ * @brief SHA-512 initialization.  Begins a SHA-512 operation.
+ *
+ * @param ctx  sha512_context_t handle to init
+ */
+void sha512_init(sha512_context_t *ctx);
+
+/**
+ * @brief Add bytes into the hash
+ *
+ * @param ctx      sha512_context_t handle to use
+ * @param[in] data Input data
+ * @param[in] len  Length of @p data
+ */
+static inline void sha512_update(sha512_context_t *ctx, const void *data, size_t len)
+{
+    sha512_common_update(ctx, data, len);
+}
+
+/**
+ * @brief SHA-512 finalization.  Pads the input data, exports the hash value,
+ * and clears the context state.
+ *
+ * @param ctx    sha512_context_t handle to use
+ * @param digest resulting digest, this is the hash of all the bytes
```

Comment:
Do we want to hint the minimum length for this buffer?
```suggestion
 * @param digest resulting digest, this is the hash of all the bytes. Must be at least XXX long.
```